app/controllers/snippets-management/index.js

```diff
@@ -0,0 +1,37 @@
+import Controller from '@ember/controller';
+import { service } from '@ember/service';
+import { task } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class SnippetsManagementIndexController extends Controller {
+  @service store;
+  @service router;
+  @service currentSession;
+  queryParams = ['page', 'size', 'label', 'sort'];
+
+  @tracked page = 0;
+  @tracked size = 20;
+  @tracked label = '';
+  @tracked sort = '-created-on';
+  @tracked isRemoveModalOpen = false;
+  @tracked deletingSnippetList;
+
+  removeSnippetList = task(async () => {
+    this.deletingSnippetList.deleteRecord();
+    await this.deletingSnippetList.save();
+    this.closeRemoveModal();
+  });
```

Comment:
- Bug for deleting a snippet list, where snippets themselves don't delete, is still present. (BUG: When deleting a snippet list, the snippets are not deleted. Reproduce by creating snippet list, add one snippet, then delete the snippet list. The snippet itself will still be in the database.)